pennylane/templates/subroutines/permute.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``permute`` template.
+"""
+
+import pennylane as qml
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.wires import Wires
+
+
+@template
+def Permute(permutation, wires):
+    r"""Permutes a set of wires into a new order.
```

Comment:
This is going to be an annoying comment because I can't think of a better alternative, but 'into a new order' sounds a bit strange.

Reply:
Updated this to something clearer :+1: